etc/kayobe/kolla/globals.yml

```diff
@@ -69,5 +69,131 @@ prometheus_ceph_mgr_exporter_endpoints:
 {% endfor %}
 {% endif %}
 
-#############################################################################
+
+########################################
+# Prometheus blackbox monitoring options
+########################################
+{% raw %}
+kolla_internal_endpoint: "{{ internal_protocol }}://{{ kolla_internal_fqdn | put_address_in_context('url') }}"
+kolla_public_endpoint: "{{ public_protocol }}://{{ kolla_external_fqdn | put_address_in_context('url') }}"
+
+prometheus_blackbox_exporter_endpoints: "{{ prometheus_blackbox_exporter_endpoints_default | selectattr('enabled', 'true') | map(attribute='endpoints') | flatten | union(prometheus_blackbox_exporter_endpoints_custom) | unique | select }}"
+
+prometheus_blackbox_exporter_endpoints_custom: []
+
+prometheus_blackbox_exporter_endpoints_default:
+  # OpenStack endpoints
+  - endpoints:
+      - "barbican:os_endpoint:{{ kolla_public_endpoint }}:{{ barbican_api_public_port }}"
+      - "{{ ('barbican_internal:os_endpoint:' + kolla_internal_endpoint + ':' + barbican_api_port) if kolla_public_endpoint != kolla_internal_endpoint }}"
+    enabled: "{{ enable_barbican | bool }}"
+  - endpoints:
+      - "blazar:os_endpoint:{{ kolla_public_endpoint }}:{{ blazar_api_public_port }}"
+      - "{{ ('blazar_internal:os_endpoint:' + kolla_internal_endpoint + ':' + blazar_api_port) if kolla_public_endpoint != kolla_internal_endpoint }}"
+    enabled: "{{ enable_blazar | bool }}"
+  - endpoints:
+      - "ceph_rgw:os_endpoint:{{ kolla_public_endpoint }}:{{ ceph_rgw_public_port }}"
```

Comment:
Have you verified this works? This would hit the S3 API which I am not sure would work with the `os_endpoint` module, since it doesn't contain `versions`.

Reply:
That one is untested. I think you are right though so will try `http_2xx`

Reply:
Done